### PR TITLE
Add keycloak backend checks secret

### PR DESCRIPTION
**Pull request by TomJKing**
Updating the database through the API with the backend checks data, requires authorisation via a keycloak service account, as backend checks triggerd by lambda, not a user

Keycloak service account authorisation requires additional secret to allow for retrieval of token by client id and client secret: https://www.keycloak.org/docs/latest/server_admin/index.html

Create and expose the backend checks secret to the relevant TDR modules to allow updating of the backend checks data via lambda

### CODE REVIEW

File: modules/keycloak/ecs.tf [hunk 1]
```diff
@@ -14,17 +14,18 @@ data "template_file" "app" {
...
+    app_image                         = "nationalarchives/tdr-auth-server:${var.environment}"
+    app_port                          = local.app_port
+    app_environment                   = var.environment
+    aws_region                        = var.region
+    url_path                          = aws_ssm_parameter.database_url.name
+    username_path                     = aws_ssm_parameter.database_username.name
+    password_path                     = aws_ssm_parameter.database_password.name
+    admin_user_path                   = aws_ssm_parameter.keycloak_admin_user.name
+    admin_password_path               = aws_ssm_parameter.keycloak_admin_password.name
+    client_secret_path                = aws_ssm_parameter.keycloak_client_secret.name
```
Comment (suzannehamilton) [inline modules/keycloak/ecs.tf:26]: How much disruption would it cause to rename this parameter to distinguish it from your new client secret?

Comment (TomJKing) [reply modules/keycloak/ecs.tf:26]: Its going to need a lot of changes in a lot of places. I'm not sure its worth the effort.

Comment (suzannehamilton) [reply modules/keycloak/ecs.tf]: Ah, OK. It might be worth doing it if we find it causes confusion, but let's leave it for now.

Comment (TomJKing) [reply modules/keycloak/ecs.tf]: I've actually implemented the changes in this PR

Comment (suzannehamilton) [reply modules/keycloak/ecs.tf]: Sorry, I misunderstood! Thanks for updating the name. 👍

**Pull request merged**